metatensor-torch/src/labels.cpp

```diff
@@ -702,6 +702,30 @@ std::string LabelsHolder::repr() const {
 
 /******************************************************************************/
 
+LabelsEntryHolder::LabelsEntryHolder(TorchLabels labels, int64_t index):
+    labels_(std::move(labels))
+{
+    auto size = labels_->values().size(0);
+    if (index < -size || index >= size) {
```

Comment:
why does this only raises an error on windows?

Reply:
The other way around (`ifndef`, not `ifdef`). The constructor of IndexError taking two strings is not available on windows (it would have to be explicitly exported by torch), so in this case we let the error happen with a full backtrace in the indexing code.

Reply:
Ahh okay, I see now the comment also makes more sense!